ipn/ipnlocal/cert.go

```diff
@@ -134,15 +137,19 @@ func (b *LocalBackend) GetCertPEMWithValidity(ctx context.Context, domain string
 		if minValidity == 0 {
 			logf("starting async renewal")
 			// Start renewal in the background, return current valid cert.
-			go b.getCertPEM(context.Background(), cs, logf, traceACME, domain, now, minValidity)
+			b.goTracker.Go(func() { getCertPEM(context.Background(), b, cs, logf, traceACME, domain, now, minValidity) })
 			return pair, nil
 		}
 		// If the caller requested a specific validity duration, fall through
 		// to synchronous renewal to fulfill that.
 		logf("starting sync renewal")
 	}
 
-	pair, err := b.getCertPEM(ctx, cs, logf, traceACME, domain, now, minValidity)
+	if envknob.IsCertShareReadOnlyMode() {
+		return nil, fmt.Errorf("retrieving cached TLS certificate failed with %w, and cert store is configured in read-only mode, not attempting to issue new certificate", err)
```

Comment:
I do think the formatting for this error will be a bit hard to read as-is, because the error message won't be quoted. Convention puts the error message after a colon which will make it much easier for readers to understand the implicit hierarchy of errors:

```suggestion
		return nil, fmt.Errorf("retrieving cached TLS certificate failed, and cert store is configured in read-only mode, not attempting to issue new certificate: %w", err)
```